### PR TITLE
[otbn] Move software to sw/otbn

### PR DESCRIPTION
**Pull request by imphil**
... as agreed on in #2935. 

This also changes the build directory in the OTBN makefile to match the comment (remove the duplicated `code-snippets` directory).

### CODE REVIEW

**Comment by mcy** [comment]
Makefiles aren't allowed in the SW tree. I take it there is a plan in the short-term to migrate to Meson?

**Comment by imphil** [comment]
This PR doesn't add a Makefile to the `sw` tree, it only adjusts an existing one in the `hw` tree (which is a bit of a stop-gap while we develop more complete tooling for OTBN.)

File: sw/otbn/code-snippets/README.md [hunk 1]
```diff
@@ -4,3 +4,4 @@ This directory contains some code snippets that give examples of how
 to do various tasks in OTBN code.
 
   - `modexp.S`: An example of how to do modular exponentiation.
+  - `pseudo-ops.S`: An example of the pseudo-operations supported by the OTBN ISA.
```
Comment (rswarbrick) [inline sw/otbn/code-snippets/README.md:7]: Oops, sorry for missing this!

**Comment by sriyerg** [comment]
> This PR doesn't add a Makefile to the `sw` tree, it only adjusts an existing one in the `hw` tree (which is a bit of a stop-gap while we develop more complete tooling for OTBN.)

The DVSim flow supports meson for building SW for the Ibex. It will be great if we could leverage the same framework for OTBN IP level / chip level DV.

**Comment by imphil** [comment]
> It will be great if we could leverage the same framework for OTBN IP level / chip level DV.

We will, but we're not there yet (see #2936). There are currently many moving parts in the OTBN software toolchain, and the integration into the meson-based toolchain will come as soon as we actually have tools such as a compiler, linker, etc. sufficiently stable. 

This PR is only moving OTBN application snippets into the `sw` tree, all build-system related issues are not really in scope for this PR.

**Pull request merged**